Lesson21/nurbasak/script.js

```diff
@@ -0,0 +1,168 @@
+    nameError.classList.remove('hidden');
+    formValid = false;
+  } else if (!(name.trim().length > 0)) {
+    nameError.classList.remove('hidden');
+    nameError.innerText = 'Enter more than one character.';
+    formValid = false;
+  } else {
+    nameError.innerText = '';
+    nameError.classList.add('hidden');
+  }
+
+  nameInput.addEventListener('change', () => {
+    nameOutput.textContent = nameInput.value;
+    data.name = nameInput.value;
+    allowSubmit();
+  })
+};
+
+nameInput.addEventListener('input', () => validateName(nameInput.value, nameInputError));
+
+function validateEmail(email,emailError) {
+  if (!emailPattern.test(email)) {
+    emailError.innerText = 'Please enter a valid email address.';
+    emailError.classList.remove('hidden');
+    formValid = false;
+  } else {
+    emailError.innerText = '';
+    emailError.classList.add('hidden');
+  }
+  emailInput.addEventListener('change', () => {
+    emailOutput.textContent=emailInput.value;
+    data.email = emailInput.value;
+    allowSubmit();
+  })
+};
+
+emailInput.addEventListener('input', () => validateEmail(emailInput.value, emailInputError));
+
+//added functionality to show date line on click if supported
+
+dateInput.addEventListener('click',() => {
+  if(typeof dateInput.showPicker === 'function'){
+    dateInput.showPicker();
+  }
+})
+
+//added min date functionality. But it is help with ai and I thought I found easier way but it didnt work at all
+
+function getTomorrowDate() {
+  const today = new Date();
+  const tomorrow = new Date(today);
+  tomorrow.setDate(tomorrow.getDate() + 1);
+  return tomorrow;
+}
+
+function setMinDateToTomorrow() {
+  const t = getTomorrowDate();
+  const yyyy = t.getFullYear();
+  const mm = String(t.getMonth() + 1).padStart(2, '0');
+  const dd = String(t.getDate()).padStart(2, '0');
+  const minDate = `${yyyy}-${mm}-${dd}`;
+
+  dateInput.min = minDate;
+
+  if (!dateInput.value || dateInput.value < minDate) {
+    dateInput.value = minDate;
+    dateOutput.textContent = dateInput.value;
+    data.date = dateInput.value;
+  }
+}
+
+setMinDateToTomorrow();
+
+dateInput.addEventListener('change', () => {
+  dateOutput.textContent = dateInput.value;
+  data.date = dateInput.value;
+
+  allowSubmit();
+});
+
+//added deselect after new selection function for timeslot buttons
+
+timeslotButtons.forEach(function (element) {
+  element.addEventListener('click', () => {
+    timeslotButtons.forEach(slot => {
+      slot.classList.remove('selected');
+    });
+    element.classList.add('selected');
+    showSelectedTime(element);
+  });
+});
+
+//ADD MIN DATE FUNCTIONALITY
+
+function showSelectedTime(element) {
+  element.classList.add('selected');
+  // TODO: deselect element if another one is selected
+  selectedTimeOutput.textContent = element.textContent;
+  data.time = element.textContent;
+  allowSubmit();
+}
+
+function allowSubmit() {
+  if (data.date && data.time && data.name && data.email) {
```

Comment:
The check `if (!data.date && !data.time && !data.name && !data.email) return;` only prevents submission if all fields are empty, not if one is missing. Use logical OR (||) instead of AND (&&).